### PR TITLE
openshot-qt: enable on darwin

### PR DESCRIPTION
**Pull request by veprbl**
###### Motivation for this change

Provide OpenShot on darwin

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/openshot-qt/default.nix [hunk 1]
```diff
@@ -47,6 +54,6 @@ mkDerivationWith python3Packages.buildPythonApplication rec {
     '';
     license = with licenses; gpl3Plus;
     maintainers = with maintainers; [ AndersonTorres ];
-    platforms = with platforms; linux;
+    platforms = with platforms; darwin ++ linux;
```
Comment (SuperSandro2000) [inline pkgs/applications/video/openshot-qt/default.nix:57]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/applications/video/openshot-qt/libopenshot-audio.nix [hunk 2]
```diff
@@ -15,8 +19,12 @@ stdenv.mkDerivation rec {
   nativeBuildInputs =
   [ pkgconfig cmake doxygen ];
 
-  buildInputs =
-  [ alsaLib libX11 libXft libXrandr libXinerama libXext libXcursor ];
+  buildInputs = optionals stdenv.isLinux [ alsaLib ]
+    ++ (if stdenv.isDarwin then
+      [ zlib AGL Cocoa Foundation ]
+    else
```
Comment (SuperSandro2000) [inline pkgs/applications/video/openshot-qt/libopenshot-audio.nix:25]: Can you use if stdenv.isLinux?

Comment (veprbl) [reply pkgs/applications/video/openshot-qt/libopenshot-audio.nix]: Sure.

File: pkgs/applications/video/openshot-qt/default.nix (hunk 1 above)
Comment (veprbl) [reply pkgs/applications/video/openshot-qt/default.nix:57]: I don't think it will work with BSD sound system.

**Comment by AndersonTorres** [comment]
There is a FreeBSD port for openshot:

https://www.freshports.org/multimedia/openshot

File: pkgs/applications/video/openshot-qt/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/applications/video/openshot-qt/default.nix:57]: @veprbl But there is a FreeBSD port for openshot:

https://www.freshports.org/multimedia/openshot

Comment (veprbl) [reply pkgs/applications/video/openshot-qt/default.nix]: Fair enough. Changed to `unix`.

Comment (AndersonTorres) [reply pkgs/applications/video/openshot-qt/default.nix]: Well, as far as I know we don't have FreeBSD testers yet.
Maybe I need to restart that old project...

**Pull request merged**